pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -84,6 +84,24 @@ const (
 	TemplateNAALabel      = "volume.csi.k8s.io/affinity-source-naa"
 )
 
+var labelValueSanitizer = regexp.MustCompile(`[^A-Za-z0-9\-_.]`)
+
+func sanitizeLabelValue(value string) string {
```

Comment:
I think there must be a kubernetes function for that,

Reply:
They have validation functions in the lib you sent, I did not see sanitization ones.